#17, #21, #22, #23, #27, #31 Cleaning inline styles,fixing logout rou…

pinkdragon1000 opened the pull request:
Cleaning inline styles, changing px to rem, fixing logout route, and adding button and card components

pinkdragon1000 committed:
#17, #21, #22, #23, #27, #31 Cleaning inline styles,fixing logout route,adding button and card components

Cleaning inline styles, changing px to rem, fixing logout route,adding button and card components

pinkdragon1000 committed:
#31 changing some missed px to rem

msdominguez commented on frontend/src/App.js:
```diff
@@ -23,8 +23,21 @@ import SideNav, { NavItem, NavIcon } from '@trendmicro/react-sidenav';
+
 const LoginContainer = () => (
-  <div className="App-header" style={{ backgroundColor: 'white' }}>
+  <div className="App-header loginBackground">
```
is this a style that's shared btwn client and advisor and thats why its in styles.css instead of a const in this file?

pinkdragon1000 replied:
**Just changed this to use const**

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
 
-  <div className="App-header" style={{ backgroundColor: '#CCCCCC' }}>
+const ClientContainer = () => (
+  <div className="App-header clientadvisorBackground">
```
is this a style that's shared btwn client and advisor and thats why its in styles.css instead of a const in this file?

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
+                    <NavIcon>
                       <FontAwesomeIcon icon={faUser} />
-                      <span style={{ fontSize: '10px', position: 'absolute', top: '15px', left: '17px' }}>Profile</span>
+                      <span style={{ fontSize: '0.625rem', position: 'absolute', top: '0.9375rem', left: '1.0625rem' }}>Profile</span>
```
this should be a const css at the top of this file but nice job on the px -> rem

pinkdragon1000 replied:
**Just changed this to use const**

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
                     <NavIcon>
                       <FontAwesomeIcon icon={faChartBar} />
-                      <span style={{ fontSize: '10px', position: 'absolute', top: '15px', left: '22px' }}>Plan</span>
+                      <span style={{ fontSize: '0.625rem', position: 'absolute', top: '0.9375rem', left: '1.375rem' }}>Plan</span>
```
this should be a const css at the top of this file but nice job on the px -> rem

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
                     <NavIcon>
                       <FontAwesomeIcon icon={faCheckCircle} />
-                      <span style={{ fontSize: '10px', position: 'absolute', top: '15px', left: '4px' }}>Action Items</span>
+                      <span style={{ fontSize: '0.625rem', position: 'absolute', top: '0.9375rem', left: '0.25rem' }}>Action Items</span>
```
this should be a const css at the top of this file but nice job on the px -> rem

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
                     <NavIcon>
                       <FontAwesomeIcon icon={faAddressCard} />
-                      <span style={{ fontSize: '10px', position: 'absolute', top: '15px', left: '4px' }}>Advisor Contact</span>
+                      <span style={{ fontSize: '0.625rem', position: 'absolute', top: '0.9375rem', left: '0.25rem' }}>Advisor Contact</span>
```
this should be a const css at the top of this file but nice job on the px -> rem

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
-                    </NavIcon>
-                  </NavItem>
-
+                      <span style={{ fontSize: '0.625rem', position: 'absolute', top: '0.9375rem', left: '1.0625rem' }}>Logout</span>
```
this should be a const css at the top of this file but nice job on the px -> rem

msdominguez commented on frontend/src/index.css:
```diff
@@ -13,82 +11,67 @@ body {
 
-.h3 {
-    font-family: 'Libre Baskerville', serif;
+.clientadvisorBackground{
```
lets try to make all our css class names hyphenated and not camelCase (i see a few diff ones in here so lets just change them all earlier than later)

pinkdragon1000 replied:
okay I'll do that!

msdominguez commented on frontend/src/index.css:
```diff
@@ -13,82 +11,67 @@ body {
 
-.p-4 {
-    padding: 5rem!important;
+.loginBackground{
```
lets try to make all our css class names hyphenated and not camelCase (i see a few diff ones in here so lets just change them all earlier than later)

msdominguez commented on frontend/src/index.css:
```diff
@@ -13,82 +11,67 @@ body {
+    word-wrap: break-word;
+    white-space: normal;
+    cursor: pointer;
+    border: 0;
```
ive never seen someone use "border: 0;", shouldnt it be "border: none" ? sorry this is nitpicking but want to keep things consistent

pinkdragon1000 replied:
**Changed to border:none**

msdominguez commented on frontend/src/index.css:
```diff
@@ -13,82 +11,67 @@ body {
 
-.fontawesomeicon.blue-purple-gradient {
-    color: linear-gradient(45deg, rgb(114, 30, 190), rgb(50, 96, 179));
+.spacerPrimary{
```
lets try to make all our css class names hyphenated and not camelCase (i see a few diff ones in here so lets just change them all earlier than later)

msdominguez commented on frontend/src/index.css:
```diff
@@ -13,82 +11,67 @@ body {
 
-.col.col-md-6 {
-    padding-top: 5rem;
+.spacerSecondary{
```
lets try to make all our css class names hyphenated and not camelCase (i see a few diff ones in here so lets just change them all earlier than later)

msdominguez commented on frontend/src/routes/ClientScreens/ClientPlan.js:
```diff
@@ -78,9 +69,8 @@ class ClientPlan extends Component {
-    function Step1(props) {
-      var incomeval = 87000;
 
+    function EmergencySavings(props) {
```
i just noticed that u make ur functions start with capital letters, is this a react thing? cuz i have always used camelCase for functions lol

pinkdragon1000 replied:
**The way I understand how this works is that Emergency Savings in this example is treated like a functional component.  In React there is the concept of functional components and class components: https://medium.com/@Zwenza/functional-vs-class-components-in-react-231e3fbd7108.  Check out line 367 in ClientPlan.js: `<EmergencySavings currentStep={this.state.currentStep} />`.  
I am just keeping the styles common between "component elements".  For example the button components created were also uppercase for example SecondaryButton.  A common convention for component names as I understand it is to use uppercase characters: https://medium.com/@wittydeveloper/react-components-naming-convention-%EF%B8%8F-b50303551505.

Here is a link to how I implemented the multistep form.  It explains a little bit more how it works: https://css-tricks.com/the-magic-of-react-based-multi-step-forms/**

msdominguez commented on frontend/src/routes/Components/SecondaryButton.js:
```diff
@@ -0,0 +1,15 @@
+      backgroundColor: "#0364FF",
+      borderRadius: "0.625rem",
+      bottom:"0rem",
+      right:"15rem"
```
what do these do? is this to center the text? or add padding? cuz i think i wouldve done "padding: 1rem" or something and "text-align: center"

msdominguez commented on frontend/src/routes/Components/QuaternaryButton.js:
```diff
@@ -0,0 +1,19 @@
+      borderRadius: "0.625rem",
+      border: "0.0625rem solid #0364FF",
+      bottom:"0rem",
+      right:"22rem"
```
what do these do? is this to center the text? or add padding? cuz i think i wouldve done "padding: 1rem" or something and "text-align: center"

pinkdragon1000 replied:
**This wasn't needed so I removed it.**

msdominguez commented on frontend/src/routes/Components/PrimaryButton.js:
```diff
@@ -0,0 +1,17 @@
+      borderRadius: "0.625rem",
+      bottom:"0rem",
+      right:"15rem",
+      float:"right"
```
what do these do? is this to center the text? or add padding? cuz i think i wouldve done "padding: 1rem" or something and "text-align: center"

msdominguez commented on frontend/src/routes/Components/TertiaryButton.js:
```diff
@@ -0,0 +1,17 @@
+      borderRadius: "0.625rem",
+      border: "0.05rem #949494 solid",
+      color: "#949494",
+      minWidth: "12rem",
```
i don't understand why this button's code is not similar to the other buttons.. how come it has a min width but the others don't? + also what about padding/text-align center?

pinkdragon1000 replied:
**Added minWidth to other button elements**

msdominguez commented on frontend/src/routes/Login.js:
```diff
@@ -16,13 +19,28 @@ class Login extends Component {
     };
 
+
+    const paddingTop = {
```
this name tells me nothing of what this class is for (like what elements)

pinkdragon1000 replied:
My bad I'll fix it!

pinkdragon1000 replied:
**Changed to `signinSpacer`.  It is just a spacer so the Sign In text starts after the curve of the gradient on the left side.  I don't know a better name for this so I think it's good enough.**

msdominguez commented on frontend/src/App.js:
```diff
@@ -33,73 +46,75 @@ const LoginContainer = () => (
                   </NavItem>
 
-                  <div style={{ paddingBottom: '40vh' }}></div>
+                  <div style={paddingBottom}></div>
```
pls make sure this css class is renamed

msdominguez commented on frontend/src/App.js:
```diff
@@ -23,8 +23,21 @@ import SideNav, { NavItem, NavIcon } from '@trendmicro/react-sidenav';
+  paddingBottom: '0.3125rem'
+}
+
+const paddingBottom={
```
this class name does not tell me what elements is used for.. even like "spacer" is a better name. just dont name it based on the attributes cuz what if u want to add more/this is just bad practice?

pinkdragon1000 replied:
**Changed it to `signoutSpacer` it's just a spacer between the sign out button and the rest of the buttons in the navbar we probably won't need to add anything more to this but I changed the name.**

pinkdragon1000 committed:
Updates based on comments in PR #33.

Cleaning up inline styles #27 and cleaning up button components to be more consistent with each other (#21 , #22, #23).

pinkdragon1000 committed:
updating package json files
